### PR TITLE
add conda-index recipe

### PR DESCRIPTION
**Pull request by jharlow-intel**
Let me know if I missed anything or need to do anything else to get this approved & merged. 

Thanks in advance!

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/conda-index/meta.yaml
Comment (dholth) [inline recipes/conda-index/meta.yaml:23]: This comment can be removed

**Review by dholth — APPROVED** [review]
Looks good. We can remove the # 3.9? comment as 3.7 is supported now.

**Comment by AndresGuzman-Ballen** [comment]
@dholth when you were working on the recipe at your end, did it also work on Windows? I'm having a hard time figuring out why conda-forge's CI encountered an error on it; nothing problematic is standing out :(

File: recipes/conda-index/meta.yaml
Comment (dholth) [inline recipes/conda-index/meta.yaml:15]:
```suggestion
  noarch: python
  entry_points:
    - conda-index = conda_index.cli:cli
```

Try that

**Comment by jharlow-intel** [comment]
@dholth looks like `pytest -k test_index_on_single_subdir_1` isn't working for the windows CI, for whatever reason

**Comment by dholth** [comment]
Here's the problem `E       assert b'{\r\n  "inf...": 1\r\n}\r\n' == b'{\n  "info"...sion": 1\n}\n'`

We will need to make sure conda-index uses `\n` instead of `\r\n` on all platforms.

**Comment by dholth** [comment]
https://github.com/conda/conda-index/issues/74

**Comment by AndresGuzman-Ballen** [comment]
Oh Windows...so good at being the worst

**Comment by dholth** [comment]
https://github.com/conda/conda-index/pull/75

File: recipes/conda-index/meta.yaml
Comment (dholth) [inline recipes/conda-index/meta.yaml:2]:
```suggestion
{% set version = "0.2.2" %}
```

File: recipes/conda-index/meta.yaml
Comment (dholth) [inline recipes/conda-index/meta.yaml:10]:
```suggestion
  sha256: 07ac9ed0938b88a02f093644e4ab0fa41919fec87fa2e1b2f29a761a6e3f2953
```

**Comment by jakirkham** [comment]
Isn't this [part of `conda-build` currently]( https://github.com/conda-forge/conda-build-feedstock/blob/dbeaa577de8470bcf5f93bd3035daf3474d8257e/recipe/meta.yaml#L19 )? Should we make them [conflict somehow]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#defining-conflicts )?

**Comment by jakirkham** [comment]
cc @conda-forge/core

**Comment by beckermr** [comment]
Yes they need to conflict with versions of conda build that have the code.

**Review by beckermr — CHANGES_REQUESTED** [review]
See comments above on conda build.

**Comment by beckermr** [comment]
We need @dholth and co to figure this out before we merge this IMHO. I am willing to be convinced otherwise though.

**Comment by jharlow-intel** [comment]
the test case `test_index_on_single_subdir_1` does have a conda-build dependency too so we'll have to work it out somehow.

**Comment by dholth** [comment]
conda-index and conda-build both provide the `conda-index` script. I've been thinking about renaming it to something else? Any suggestions? conda-index-2 doesn't roll off the tongue.

Apart from that, both software should work correctly when installed together, with conda-build continuing to call its internal index through its API.

In about six months we will make conda-build depend on conda-index, and remove conda-build's internal index support.

File: recipes/conda-index/meta.yaml
Comment (dholth) [inline recipes/conda-index/meta.yaml:36]:
```suggestion
```

They should be installed at the same time. Do we need to remove the console script from conda-index or rename?

**Comment by dholth** [comment]
This PR simply removes the conda-index script until we figure out a better idea. https://github.com/conda/conda-index/pull/76/files

File: recipes/conda-index/meta.yaml
Comment (dholth) [inline recipes/conda-index/meta.yaml:17]: Remove the entry point section (see upstream)

Comment (jharlow-intel) [reply recipes/conda-index/meta.yaml:17]: oops, good catch. Thanks

**Review by dholth — APPROVED** [review]
👍

File: recipes/conda-index/meta.yaml
Comment (jakirkham) [inline recipes/conda-index/meta.yaml:36]: Could you please clarify what this selector is doing?

Comment (dholth) [reply recipes/conda-index/meta.yaml:36]: This is a mistake. Coverage's toml support should be installed.

Comment (jakirkham) [reply recipes/conda-index/meta.yaml:36]: Gotcha. So this is related to issue ( https://github.com/conda/conda/issues/2984 ) and a couple others ( https://github.com/conda/conda/issues/2984#issuecomment-1333274998 )

In any event, we don't currently have this exposed. Given this is [just `tomli`]( https://github.com/nedbat/coveragepy/blob/7e7c44b4f5c484105559690b1efccd84839bc640/setup.py#L112 ), maybe we can list that here too?

Comment (jharlow-intel) [reply recipes/conda-index/meta.yaml:36]: Just to be clear, you're suggesting remove the ` # [toml]` selector and adding `tomli` as an additional test requirement?

Comment (dholth) [reply recipes/conda-index/meta.yaml:36]: yes

Comment (dholth) [reply recipes/conda-index/meta.yaml:36]: https://github.com/nedbat/coveragepy/blob/master/setup.py#L112

**Pull request merged**
